src/vehicle.cpp

```diff
@@ -6784,71 +6774,45 @@ void vehicle::update_time( const calendar &update_to )
 /*-----------------------------------------------------------------------------
  *                              VEHICLE_PART
  *-----------------------------------------------------------------------------*/
-vehicle_part::vehicle_part( int const dx, int const dy )
-: id( NULL_ID )
-, mount( dx, dy )
-, precalc( { { point( -1, -1 ), point( -1, -1 ) } } )
-, amount( 0 )
-{
-}
+vehicle_part::vehicle_part()
+    : mount( 0, 0 ), id( NULL_ID ) {}
 
-vehicle_part::vehicle_part( const vpart_str_id &sid, int const dx, int const dy,
-                            const item *const it )
-: vehicle_part( dx, dy )
+vehicle_part::vehicle_part( const vpart_str_id& str, int const dx, int const dy, item&& obj )
+    : mount( dx, dy ), id( str ), base( std::move( obj ) )
 {
-    set_id( sid );
-    if( it != nullptr ) {
-        properties_from_item( *it );
+    const vpart_info& vp = id.obj();
+
+    if( base.typeId() != vp.item ) {
+        debugmsg( "incorrect vehicle part item, expected: %s, received: %s",
+                  vp.item.c_str(), base.typeId().c_str() );
     }
-}
 
-void vehicle_part::set_id( const vpart_str_id & str )
-{
-    id = str;
+    if( base.is_var_veh_part() ) {
+        bigness = base.bigness;
+    }
+
+    // item damage is [ 0..MAX_ITEM_DAMAGE ] whereas part hp is [ 1..durability ]
+    int health = ( MAX_ITEM_DAMAGE + 1 ) - base.damage;
+    health *= vp.durability; // [ 0, durability ]
+    health /= ( MAX_ITEM_DAMAGE + 1 );
+    hp = std::max( 1, health );
+
+    if( vp.fuel_type == fuel_type_battery ) {
```

Comment:
All other parts have their ammo ejected before they are installed so the code is dead